### PR TITLE
[control-plane-manager] Calculate and add etcd quota-backend-bytes parameter

### PR DESCRIPTION
**Pull request by name212**
## Description
Add hook for calculation quota-backend-bytes etcd parameter and set this parameter.

## Why do we need it, and what problem does it solve?
Big Kubernetes clusters have more data stored in etcd.
If `quota-backend-bytes` is exceeded, then etcd clusters goes in error state and only get and delete operations are allowed. This leads to degraded Kubernetes cluster performance.

## Changelog entries

```changes
section: control-plane-manager
type: feature
summary: Calculate and add etcd quota-backend-bytes parameter
impact: |
  Etcd will be restarted. Will add quota-backend-bytes parameter. This parameter will calculate depends on control-plane memory capacity.
impact_level: high
```

### CODE REVIEW

File: modules/040-control-plane-manager/monitoring/prometheus-rules/etcd-maintenance.yaml
Comment (yalosev) [inline modules/040-control-plane-manager/monitoring/prometheus-rules/etcd-maintenance.yaml:67]:
```suggestion
          It happens when control-plane nodes memory was reduced.
```

File: modules/040-control-plane-manager/hooks/etcd_defragmentation.go
Comment (name212) [reply modules/040-control-plane-manager/hooks/etcd_defragmentation.go]: we removed auto defragmentation :-)

File: modules/340-monitoring-kubernetes-control-plane/monitoring/prometheus-rules/kube-etcd3.yaml
Comment (nabokihms) [reply modules/340-monitoring-kubernetes-control-plane/monitoring/prometheus-rules/kube-etcd3.yaml:118]: We agreed that this is not important because if Deckhouse does not manage etcd (e.g., managed cloud services), it can do nothing about the size of the database.

**Review by nabokihms — DISMISSED** [review]

File: modules/040-control-plane-manager/monitoring/prometheus-rules/etcd-maintenance.yaml
Comment (yalosev) [inline modules/040-control-plane-manager/monitoring/prometheus-rules/etcd-maintenance.yaml:11]: don't you want to move it to `for: 10m` instead of polk annotation?

Comment (name212) [reply modules/040-control-plane-manager/monitoring/prometheus-rules/etcd-maintenance.yaml:11]: I don't know :-) This alert was moved from [monitoring-kubernetes-control-plane module](https://github.com/deckhouse/deckhouse/pull/1389/files/ff50c53fa2a0f8c9a9b1b5df058f7ecc43a3e021#diff-490f6a44424e715726f88cfda3b952a536093e8e93da155fa1be8ffe0ea57590L118) with change in expression.

Comment (name212) [reply modules/040-control-plane-manager/monitoring/prometheus-rules/etcd-maintenance.yaml:11]: @nabokihms What do you thing about it?

File: modules/040-control-plane-manager/monitoring/prometheus-rules/etcd-maintenance.yaml
Comment (konstantin-axenov) [inline modules/040-control-plane-manager/monitoring/prometheus-rules/etcd-maintenance.yaml:40]:
```suggestion
        d8_module: control-plane-manager
        severity_level: "6"
```

File: modules/040-control-plane-manager/monitoring/prometheus-rules/etcd-maintenance.yaml
Comment (konstantin-axenov) [reply modules/040-control-plane-manager/monitoring/prometheus-rules/etcd-maintenance.yaml:11]: Yes, you should use `for`. We are making such changes for other alerts in https://github.com/deckhouse/deckhouse/pull/1446

Comment (name212) [reply modules/040-control-plane-manager/monitoring/prometheus-rules/etcd-maintenance.yaml]: done

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]

File: modules/040-control-plane-manager/hooks/common_test.go
Comment (name212) [reply modules/040-control-plane-manager/hooks/common_test.go:139]: done

File: modules/040-control-plane-manager/openapi/doc-ru-config-values.yaml
Comment (name212) [reply modules/040-control-plane-manager/openapi/doc-ru-config-values.yaml:130]: done

File: modules/040-control-plane-manager/hooks/common.go
Comment (name212) [reply modules/040-control-plane-manager/hooks/common.go:46]: I removed all consts with snapshot name.
And I thing name of snapshot is unnecessary here.

File: modules/040-control-plane-manager/hooks/common_test.go
Comment (name212) [reply modules/040-control-plane-manager/hooks/common_test.go]: moved

**Review by RomanenkoDenys — APPROVED** [review]

**Review by nabokihms — APPROVED** [review]
LGTM! We are waiting for @konstantin-axenov and @yalosev for the final decision.

**Review by yalosev — APPROVED** [review]

**Pull request merged**
